frontends/PyCDE/test/generator_options.py

```diff
@@ -0,0 +1,43 @@
+# RUN: %PYTHON% %s | FileCheck %s
+
+from pycde import System, module, generator, types
+
+from circt.dialects import hw
+
+
+@module
+class GeneratorOptions:
+
+  @generator
+  def generator_a(mod):
+    hw.ConstantOp.create(types.i32, 1)
+
+  @generator
+  def generator_b(mod):
+    hw.ConstantOp.create(types.i32, 2)
+
+
+class Top(System):
+  inputs = []
+  outputs = []
+
+  def build(self, top):
+    GeneratorOptions()
+
+
+# CHECK: hw.constant 1
+top1 = Top()
+top1.generate()
```

Comment:
I would prefer this be an error case. Since the order in which generators appear in the list of generators isn't well defined, the generator which gets run could change from time to time.

Alternatively, we could define a default. I'm good either way.

Reply:
How about if there is exactly one registered generator, it's not required to pass the list of generator names (since it would be redundant). This specific case would become an error.

Reply:
Yes, that's what I mean -- this one specific case (where there is >1 generator) would be an error. I would like the list of generator names to be optional except in that case.